src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,244 @@
+            );
+
+        private static readonly ImmutableArray<SyntaxKind> s_syntaxKindsConsideredAsStatementInvokesCallee =
+            ImmutableArray.Create(
+                SyntaxKind.DoStatement,
+                SyntaxKind.ExpressionStatement,
+                SyntaxKind.ForStatement,
+                SyntaxKind.IfStatement,
+                SyntaxKind.LocalDeclarationStatement,
+                SyntaxKind.LockStatement,
+                SyntaxKind.ReturnStatement,
+                SyntaxKind.SwitchStatement,
+                SyntaxKind.ThrowStatement,
+                SyntaxKind.WhileStatement,
+                SyntaxKind.TryStatement,
+                SyntaxKind.UsingStatement,
+                SyntaxKind.YieldReturnStatement);
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpInlineMethodRefactoringProvider() : base(CSharpSyntaxFacts.Instance, CSharpExpressionPrecedenceService.Instance)
+        {
+        }
+
+        private static bool ShouldStatementBeInlined(StatementSyntax statementSyntax)
+            => statementSyntax is ReturnStatementSyntax || statementSyntax is ExpressionStatementSyntax;
+
+        protected override bool IsSingleStatementOrExpressionMethod(SyntaxNode calleeMethodDeclarationSyntaxNode)
+        {
+            if (calleeMethodDeclarationSyntaxNode is MethodDeclarationSyntax declarationSyntax)
+            {
+                var blockSyntaxNode = declarationSyntax.Body;
+                // 1. If it is an ordinary method with block
+                if (blockSyntaxNode != null)
+                {
+                    var blockStatements = blockSyntaxNode.Statements;
+                    return blockStatements.Count == 1 && ShouldStatementBeInlined(blockStatements[0]);
+                }
+                else
+                {
+                    // 2. If it is an Arrow Expression
+                    var arrowExpressionNodes = declarationSyntax.ExpressionBody;
+                    return arrowExpressionNodes != null;
+                }
+            }
+
+            return false;
+        }
+
+        protected override IParameterSymbol? GetParameterSymbol(SemanticModel semanticModel,ArgumentSyntax argumentSyntaxNode, CancellationToken cancellationToken)
+            => argumentSyntaxNode.DetermineParameter(semanticModel, allowParams: true, cancellationToken);
+
+        protected override SyntaxNode? GetInlineStatement(SyntaxNode calleeMethodDeclarationSyntaxNode)
+        {
+            var declarationSyntax = (MethodDeclarationSyntax)calleeMethodDeclarationSyntaxNode;
+            SyntaxNode? inlineSyntaxNode = null;
+            var blockSyntaxNode = declarationSyntax.Body;
+            // 1. If it is a ordinary method with block
+            if (blockSyntaxNode != null)
+            {
+                var blockStatements = blockSyntaxNode.Statements;
+                if (blockStatements.Count == 1)
```

Comment:
when is this not a count of 1?

Reply:
Removed because all the unwanted cases are filtered beforehand